crates/aranya-client-capi/src/api/defs.rs

```diff
@@ -1353,6 +1353,58 @@ pub unsafe fn remove_sync_peer(
     Ok(())
 }
 
+/// Subscribe to hello notifications from a peer.
+///
+/// This will request the peer to send hello notifications when their graph head changes.
+/// The `delay_milliseconds` parameter specifies the minimum delay between notifications.
+///
+/// @param[in] client the Aranya Client [`Client`].
+/// @param[in] team the team's ID [`TeamId`].
+/// @param[in] addr the peer's Aranya network address [`Addr`].
+/// @param[in] delay_milliseconds minimum delay between notifications in milliseconds.
+///
+/// @relates AranyaClient.
+pub unsafe fn hello_subscribe(
```

Comment:
We should use `sync` in the name of these new methods to specify this is related to the syncer.
```suggestion
/// Subscribe to hello notifications from a sync peer.
///
/// This will request the peer to send hello notifications via the sync protocol when their graph head changes.
/// The `delay_milliseconds` parameter specifies the minimum delay between notifications.
///
/// @param[in] client the Aranya Client [`Client`].
/// @param[in] team the team's ID [`TeamId`].
/// @param[in] addr the peer's Aranya network address [`Addr`].
/// @param[in] delay_milliseconds minimum delay between notifications in milliseconds.
///
/// @relates AranyaClient.
pub unsafe fn sync_hello_subscribe(
```